content/en/blog/_posts/2021-04-15-local-storage-features-go-beta.md

```diff
@@ -0,0 +1,164 @@
+---
+layout: blog
+title: 'Local Storage: Storage Capacity Tracking, Distributed Provisioning and Generic Ephemeral Volumes hit Beta
+date: 2021-04-15
+slug: 2021-04-15-local-storage-features-go-beta
+---
+
+ **Authors:** Patrick Ohly (Intel)
+
+Several features are getting promoted to beta in Kubernetes 1.21 and
+the associated CSI external-provisioner release that simplify the
+development of Container Storage Interface (CSI) drivers which manage
+storage locally on a node.
+
+This blog post explains how such drivers worked before and how these
+features help make drivers simpler.
+
+## The good...
+
+There are drivers for local storage, like TopoLVM for traditional
+disks and PMEM-CSI for persistent memory. They work and are ready for
+usage today also on older Kubernetes release, but making that possible
+was not trivial.
+
+The first problem is volume provisioning: it is handled through the
+Kubernetes control plane. Some central component must react to
+persistent volume claims (PVC) and create volumes. Usually, that is
+handled by a central deployment of the external-provisioner and a CSI
+driver component that then connects to the storage backplane. But for
+local storage, there is no such backplane.
+
+TopoLVM solved this by having its different components communicate
+with each other through the Kubernetes API server by creating and
+reading custom resources. So although TopoLVM is based on CSI, a
+standard that is independent of a particular container orchestrator,
+TopoLVM only works on Kubernetes.
+
+PMEM-CSI created its own storage backplane with communication through
+gRPC calls. Securing that communication depends on TLS certificates,
+which made driver deployment more complicated.
+
+The next problem is scheduling. When volumes get created independently
+of pods ("immediate binding"), the CSI driver must pick a node without
+knowing anything about the pod(s) that are going to use it. Topology
+information then forces those pods to run on the node where the volume
+was created. If other resources like RAM or CPU are exhausted there,
+the pod cannot start. This can be avoided by configuring in the
+storage class that volume creation is meant to wait for the first pod
```

Comment:
I'd write:
```suggestion
StorageClass that volume creation is meant to wait for the first pod
```
to highlight that StorageClass is an API and an object kind in Kubernetes.

Reply:
Also for "persistent volume claim", I suppose. I changed both.